### PR TITLE
feat: [sc-32770] User Module route updates for clark-refactor

### PR DESCRIPTION
**Pull request by Greek**
Story details: https://app.shortcut.com/clarkcan/story/32770

### CODE REVIEW

File: src/app/auth/register/components/recaptcha/recaptcha-validator.service.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export class RecaptchaValidator {
 
     validateToken(token: string) {
```
Comment (Cwagne17) [inline src/app/auth/register/components/recaptcha/recaptcha-validator.service.ts:15]: Nit: Can this be moved to the user service?

File: src/app/core/learning-object-module/editor.service.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class EditorService {
     authorUsername: string,
     status: string,
   ): Observable<string> {
-    const route = LEGACY_USER_ROUTES.UPDATE_MY_LEARNING_OBJECT(authorUsername, id);
+    const route = LEARNING_OBJECT_ROUTES.UPDATE_LEARNING_OBJECT_STATUS(id);
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/editor.service.ts:24]: Can this be moved to the learning object service?

File: src/app/core/user-module/user.service.ts [hunk 3]
```diff
@@ -95,6 +75,16 @@ export class UserService {
         );
   }
 
+  getUsersLearningObjects(username: string): Promise<LearningObject[]> {
+    return this.http
+      .get(USER_ROUTES.GET_USER(username), { withCredentials: true })
```
Comment (Cwagne17) [inline src/app/core/user-module/user.service.ts:80]: Nit: Can we update one of the names either the function or the route name. because one looks like we are getting all learning objects for a user and the other looks like we are getting the user's information.

Comment (Cwagne17) [reply src/app/core/user-module/user.service.ts:80]: I just looked at the route in the USER_ROUTES. that is the route to get the user's document in the users collection not the route to get learning objects for a user.

File: src/app/core/user-module/user.service.ts [hunk 4]
```diff
@@ -95,6 +75,16 @@ export class UserService {
...
+      .pipe(catchError(this.handleError))
+      .toPromise()
+      .then((val: any) => {
+        return val.map((l) => new LearningObject(l));
```
Comment (Cwagne17) [inline src/app/core/user-module/user.service.ts:84]: This will return a mostly empty LearningObject because it is trying to map a User to a LearningObject.

File: src/app/onion/core/learning-object.service.ts [hunk 5]
```diff
@@ -143,26 +138,6 @@ export class LearningObjectService {
...
       });
   }
-  /**
-   * Fetches Learning Object by ID (full)
-   *
-   * @param {string} learningObjectID
-   * @returns {Promise<LearningObject>}
-   * @memberof LearningObjectService
-   */
-  getLearningObject(learningObjectId: string): Promise<LearningObject> {
```
Comment (Cwagne17) [inline src/app/onion/core/learning-object.service.ts:153]: Was this not being used?

File: src/app/shared/components/user-card/user-card.component.ts [hunk 6]
```diff
@@ -25,11 +25,12 @@ export class UserCardComponent implements OnInit, OnChanges {
   }
 
   ngOnChanges() {
+    // XXX: When is this called??
     this.fetchLearningObjects();
   }
 
   async fetchLearningObjects() {
-    this.objects = await this.learningObjectService.getUsersLearningObjects(this.user.username);
+    this.objects = await this.userService.getUsersLearningObjects(this.user.username);
```
Comment (Cwagne17) [inline src/app/shared/components/user-card/user-card.component.ts:33]: This is the route that should be used for this.

This is from clark-service, it should be in the /learning-objects/search/search.service.ts

```typescript
router.get(
            "/users/:username/learning-objects",
            this.getUsersLearningObjects,
        );
```

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

File: src/app/core/user-module/user.routes.ts [hunk 7]
```diff
@@ -45,14 +45,8 @@ export const USER_ROUTES = {
...
   },
-  /**
-   * Request to retrieve a user's profile
-   * @method GET
-   * @auth required
-   * @param username the username of the user
-   * @returns the user's profile
-   */
-  GET_USER_PROFILE(username: string) {
-    return `${environment.apiURL}/users/${encodeURIComponent(username)}/profile`;
```
Comment (johngonz0) [inline src/app/core/user-module/user.routes.ts:56]: why was this removed?

**Review by johngonz0 — COMMENTED** [review]
just a question

File: src/app/core/user-module/user.routes.ts (hunk 7 above)
Comment (Greek) [reply src/app/core/user-module/user.routes.ts:56]: it's not used anywhere and i haven't seen any mention of it on the analysis docs

File: src/app/onion/core/learning-object.service.ts (hunk 5 above)
Comment (Greek) [reply src/app/onion/core/learning-object.service.ts:153]: mentioned this in the standup: this route was used when we used to fetch learning objects by their IDs, not their CUIDs. so the logic that use this function and the function itself were removed as the need for querying by ID isn't really needed anymore.

File: src/app/core/user-module/user.service.ts (hunk 3 above)
Comment (Greek) [reply src/app/core/user-module/user.service.ts]: changed to use appropriate route =)

File: src/app/core/learning-object-module/editor.service.ts (hunk 2 above)
Comment (Greek) [reply src/app/core/learning-object-module/editor.service.ts:24]: i removed the entire route because it wasn't used here, but we should probably remove the entirety of `EditorService` too because it's not referenced anywhere.

i'll keep `EditorService` for now though

File: src/app/auth/register/components/recaptcha/recaptcha-validator.service.ts (hunk 1 above)
Comment (Greek) [reply src/app/auth/register/components/recaptcha/recaptcha-validator.service.ts]: it was moved to the user service but i'm not able to test it because the Register page doesn't work, which is its only use.

**Review by stef359 — APPROVED** [review]

**Review by johngonz0 — APPROVED** [review]

File: src/app/shared/components/user-card/user-card.component.ts [hunk 8]
```diff
@@ -25,11 +25,12 @@ export class UserCardComponent implements OnInit, OnChanges {
   }
 
   ngOnChanges() {
+    // XXX: When is this called??
     this.fetchLearningObjects();
   }
```
Comment (pzalep1) [inline src/app/shared/components/user-card/user-card.component.ts:30]: ngOnChanges is a lifecycle hook. Here is an explanation: https://v17.angular.io/guide/lifecycle-hooks

**Review by pzalep1 — COMMENTED** [review]
Left a comment for the lifecycle hook explanation.

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts [hunk 9]
```diff
@@ -122,6 +137,28 @@ export const LEARNING_OBJECT_ROUTES = {
...
+          mappingsId: string,
+      ) {
+          return `${environment.apiURL}/users/${encodeURIComponent(
+              username,
+          )}/learning-objects/${encodeURIComponent(
+              learningObjectId,
+          )}/outcomes/${encodeURIComponent(outcomeId)}/mappings/${encodeURIComponent(
+              mappingsId,
+          )}`;
+      },
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:161]: Are these the paths that are used? I thought most of the routes we got rid of `/users/${username}...` Are these from standard guidelines?

File: src/app/core/user-module/user.service.ts [hunk 10]
```diff
@@ -146,6 +137,24 @@ export class UserService {
       .toPromise();
   }
 
+  /**
+   * Validate a user's captcha token
+   *
+   * @param {string} token the token to verify
+   * @returns an object with the result if fail, or null if true.
+   */
+  validateToken(token: string) {
```
Comment (Cwagne17) [inline src/app/core/user-module/user.service.ts:146]: Can we call this validate captcha instead of validate token because validate token is a separate route that we have in auth

File: src/app/core/user-module/user.routes.ts [hunk 11]
```diff
@@ -28,6 +28,15 @@ export const USER_ROUTES = {
   GET_USER(user: string) {
     return `${environment.apiURL}/users/${encodeURIComponent(user)}`;
   },
+  /**
+   * Request to retrieve a user's learning objects
+   * @method GET
+   * @param user the username or userId of the user
+   * @returns the learning objects
+   */
+  GET_USER_LEARNING_OBJECTS(user: string) {
```
Comment (Cwagne17) [inline src/app/core/user-module/user.routes.ts:37]: This route should be in the learning-object-module/search

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts [hunk 12]
```diff
@@ -122,6 +137,28 @@ export const LEARNING_OBJECT_ROUTES = {
...
+            learningObjectId,
+        )}/outcomes/${encodeURIComponent(outcomeId)}/mappings`;
+    },
+    DELETE_MAPPING(
+          username: string,
+          learningObjectId: string,
+          outcomeId: string,
+          mappingsId: string,
+      ) {
+          return `${environment.apiURL}/users/${encodeURIComponent(
```
Comment (Cwagne17) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:154]: I'm pretty sure this all got updated and should be in the learning-object-module/outcomes module. There aren't any routes called /mappings in the gateway or backend

Screenshot 2024-09-09 at 9 37 20 AM

**Review by Cwagne17 — CHANGES_REQUESTED** [review]
Some routes need to be moved to the module according to how the gateway and clark-service have them organized.

**Review by dsoto18 — APPROVED** [review]

**Review by stef359 — APPROVED** [review]

**Review by johngonz0 — APPROVED** [review]

**Review by dsoto18 — APPROVED** [review]
